src/NodaTime/Calendars/WondrousYearMonthDayCalculator.cs

```diff
@@ -0,0 +1,385 @@
+﻿// Copyright 2014 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using NodaTime.Calendars.Wondrous;
+using NodaTime.Utility;
+using System;
+
+namespace NodaTime.Calendars
+{
+    /// <summary>
+    ///     See <see cref="CalendarSystem.GetWondrousCalendar" /> for details.
+    /// </summary>
+    internal sealed class WondrousYearMonthDayCalculator : YearMonthDayCalculator
+    {
+        private const int WondrousMinYear = 1;
+        private const int WondrousMaxYear = 1000;
+
+        private const int AverageDaysPer10Years = 3652; // Ideally 365.2425 per year...
+        private const int DaysInMonth = 19;
+        private const int MonthsInYear = 19;
+
+        private const int Month18 = 18;
+        private const int Month19 = 19;
+
+        public const int AyyamiHaMonth0 = WondrousCalendarHelper.AyyamiHaMonth0;
+
+        private const int UnixEpochDayAtStartOfYear1 = -45941;
+
+        internal WondrousYearMonthDayCalculator()
+            : base(WondrousMinYear,
+                WondrousMaxYear - 1,
+                AverageDaysPer10Years,
+                UnixEpochDayAtStartOfYear1)
+        {
+            ConsoleWriteLine("~A Constructed");
+            //            System.Diagnostics.Debugger.Launch();
+        }
+
+
+        /// <summary>
+        ///     Use this to provide simple trace of calls.  Review output and sort to check test code coverage.
+        /// </summary>
+        /// <param name="format"></param>
+        /// <param name="args"></param>
+        public static void ConsoleWriteLine(string format, params object[] args)
+        {
+#if !PCL
+            Console.WriteLine(format, args);
+#endif
+        }
+
+        protected override int GetDaysFromStartOfYearToStartOfMonth(int year, int month)
+        {
+            var daysFromStartOfYearToStartOfMonth = DaysInMonth * (month - 1);
+
+            if (month == Month19)
+            {
+                daysFromStartOfYearToStartOfMonth += WondrousCalendarHelper.GetYearInfo(year).DaysInAyyamiHa;
+            }
+
+            ConsoleWriteLine("~B GetDayFromStartOfYearToStartOfMonth {0} {1} ==> {2}", year, month,
+                daysFromStartOfYearToStartOfMonth);
+            return daysFromStartOfYearToStartOfMonth;
+        }
+
+        /// <param name="wYear"></param>
+        /// <remarks>Rely on Gregorian calendar for "days" in unix epoch</remarks>
+        /// <remarks>Untested</remarks>
```

Comment:
Can you add tests for all the untested code?

Reply:
Can definitely try. Now that it seems that the calendar can be implemented in NodeTime, I can flush out some more tests. If we weren't going to be able to add it, there was no point in doing much more.